actions/marketplace.ts

```diff
@@ -3,47 +3,87 @@
 
 import {Client4} from 'mattermost-redux/client';
 
+import {getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
+import {getCurrentChannelId} from 'mattermost-redux/selectors/entities/channels';
+
 import {ActionFunc, DispatchFunc, GetStateFunc} from 'mattermost-redux/types/actions';
 
-import {getFilter, getPlugin} from 'selectors/views/marketplace';
+import type {MarketplaceApp, MarketplacePlugin} from 'mattermost-redux/types/marketplace';
+import type {CommandArgs} from 'mattermost-redux/types/integrations';
+
+import {GlobalState} from 'types/store';
+
+import {getApp, getFilter, getPlugin} from 'selectors/views/marketplace';
 import {ActionTypes} from 'utils/constants';
 
-// fetchPlugins fetches the latest marketplace plugins, subject to any existing search filter.
-export function fetchPlugins(localOnly = false): ActionFunc {
+import {executeCommand} from './command';
+
+// fetchPlugins fetches the latest marketplace plugins and apps, subject to any existing search filter.
+export function fetchListing(localOnly = false): ActionFunc {
     return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
-        const state = getState();
+        const state = getState() as GlobalState;
         const filter = getFilter(state);
 
-        try {
-            const plugins = await Client4.getMarketplacePlugins(filter, localOnly);
+        let plugins: MarketplacePlugin[];
+        let apps: MarketplaceApp[];
 
-            dispatch({
-                type: ActionTypes.RECEIVED_MARKETPLACE_PLUGINS,
-                plugins,
-            });
-
-            return {data: plugins};
+        try {
+            plugins = await Client4.getMarketplacePlugins(filter, localOnly);
         } catch (error) {
             // If the marketplace server is unreachable, try to get the local plugins only.
             if (error.server_error_id === 'app.plugin.marketplace_client.failed_to_fetch' && !localOnly) {
-                await dispatch(fetchPlugins(true));
+                await dispatch(fetchListing(true));
             }
             return {error};
         }
+
+        dispatch({
+            type: ActionTypes.RECEIVED_MARKETPLACE_PLUGINS,
+            plugins,
+        });
+
+        try {
+            apps = await Client4.getMarketplaceApps(filter);
+        } catch (error) {
+            return {data: plugins};
+        }
```

Comment:
We could fetch asynchronously here, but I drop the idea because of the added complexity. Is it worth filling a ticket for that?